mathics/builtin/specialfns/gamma.py

```diff
@@ -448,7 +447,7 @@ class Pochhammer(SympyFunction):
 
     rules = {
         "Pochhammer[0, 0]": "1",
-        "Pochhammer[a_, n_]": "Gamma[a + n] / Gamma[a]",
+        "Pochhammer[a_, n_]": "Factorial[a + n - 1] / Factorial[a - 1]",
```

Comment:
This is what `Pochhammer` should do:

```WMA
In[1]:= Pochhammer[a,b]                                                         

Out[1]= Pochhammer[a, b]

In[2]:= Pochhammer[a,12.4]                                                      

Out[2]= Pochhammer[a, 12.4]

In[3]:= Pochhammer[a,12.]                                                       

Out[3]= a (1 + a) (2 + a) (3 + a) (4 + a) (5 + a) (6 + a) (7 + a) (8 + a) 
 
>    (9 + a) (10 + a) (11 + a)

In[4]:= Pochhammer[12,b]                                                        

Out[4]= Pochhammer[12, b]

In[5]:= Pochhammer[12.1,b]                                                      

Out[5]= Pochhammer[12.1, b]
```
Notice that `n` must be an Integer to apply this rule.

```suggestion
        "Pochhammer[a_, n_Integer]": "Factorial[a + n - 1] / Factorial[a - 1]",
```

Reply:
According to the examples cited, `NumberQ`  only works when the number can be exactly converted to an `Integer`.

Reply:
Indeed. In any case, this is the point when we need to decide if the complexity of a rule that matches the exact behavior worth. The exact behavior would be obtained with something like

```
        "Pochhammer[a_, /;(IntegerQ[n]|| NumberQ[n] && Round[n]==n)]": "Factorial[a + n - 1] / Factorial[a - 1]",
```

Reply:
`IntegerQ[n]` can be dropped following the principle: simple as possible.  But even if we change to `Pochhammer[a_, n_(NumberQ[n] && Round[n]==n)]`, `Product[k, {k, i, n}]`  will transform as WMA does, in particular:  `n! / (-1+i)!`.   So WMA may be using some _other_ mechanism to implement `Product`.

But let's back up a bit.  We got into this as a result of a real problem reported by @aravindh-krishnamoorthy, presumably in his day-to-day use of Mathics3.  His PR broke something else.  If this PR addresses both his problem and does not break what is already there, then let's go with this. 

Note that this PR adds a test to ensure we do not rewrite Gamma as Factorial in the kind of way that @aravindh-krishnamoorthy was not expecting. 

In the future, if someone has a problem as a result of Pochhammer getting expanded to factorial, then we can address that issue at that time, while keeping all the other checks in place. 

In other words: complicate when there is a tangible _need_ to complicate.  Again this is also encompassed by the principle: Simple as possible, _but no simpler_.